src/psycopmlutils/synth_data_generator/synth_txt_data_generator.py

```diff
@@ -0,0 +1,167 @@
+"""Script for creating synthetic text data for testing purposes.
+
+Produces a .csv file with the following columns: citizen_id, timestamp,
+text.
+"""
+from pathlib import Path
+from typing import Dict, Iterable, List, Optional
+
+import numpy as np
+import pandas as pd
+from transformers import GPT2LMHeadModel, GPT2Tokenizer
+
+
+def generate_synth_data(
+    predictors: Dict,
+    n_samples: int,
+    text_prompt: str = "The quick brown fox jumps over the lazy dog",
+    na_prob: Optional[float] = 0.1,
+    na_ignore_cols: List[str] = [],
+) -> pd.DataFrame:
+    """Takes a dict and generates synth data from it.
+
+    Args:
+        predictors (Dict): A dict representing each column. Key is col_name (str), values are column_type (str), output_type (float|int), min (int), max(int).
+        n_samples (int): Number of samples (rows) to generate.
+        text_prompt (str): Text prompt to use for generating text data. Defaults to "The quick brown fox jumps over the lazy dog".
+        na_prob (float): Probability of changing a value in a predictor column to NA.
+        na_ignore_cols (List[str]): Columns to ignore when creating NAs
+
+    Returns:
+        pd.DataFrame: The synthetic dataset
+    """
+
+    # Initialise dataframe
+    df = pd.DataFrame(columns=list(predictors.keys()))
+
+    # Generate data
+    df = generate_data_columns(predictors, n_samples, df, text_prompt)
+
+    # randomly replace predictors with NAs
+    if na_prob:
+        mask = np.random.choice([True, False], size=df.shape, p=[na_prob, 1 - na_prob])
+        df_ = df.mask(mask)
+
+        # For all columns in df.columns if column is not in na_ignore_cols
+        for col in df.columns:
+            if col not in na_ignore_cols:
+                df[col] = df_[col]
+
+    return df
+
+
+def generate_data_columns(
+    predictors: Iterable[Dict],
+    n_samples: int,
+    df: pd.DataFrame,
+    text_prompt: str = "The quick brown fox jumps over the lazy dog",
```

Comment:
```suggestion
    text_prompt: str ,
```
More safe to only set defaults in one function and propagate from there